docs/source/programs_and_gates.rst

```diff
@@ -635,8 +635,31 @@ any other instruction:
 
    ...
 
+.. _quil_t_qvm_warning:
+
 .. warning::
-   `DELAY` and other Quil-T instructions aren't supported by the QVM.
+
+   ``DELAY`` and other Quil-T instructions are not supported by the QVM or ``quilc``. If you want to test the validity
+   of a Quil-T containing program on a QVM you should remove all Quil-T instructions before running it. You can do this 
+   dynamically by checking the ``qam`` property on your requested :py:class:`~pyquil.api.QuantumComputer`:
+
+   .. testcode:: remove-quil-t
+
+    from pyquil.quil import Program
+    from pyquil.gates import DELAY, H
+    from pyquil.api import QVM, get_qc
+
+    qc = get_qc("2q-qvm")
+    p = Program(H(0))
+    p += DELAY(0, 200e-9)
+
+    # If we're using a QVM, remove the Quil-T instructions
+    if isinstance(qc.qam, QVM):
+        p = p.remove_quil_t_instructions()
+    else: # Otherwise, compile to native Quil
```

Comment:
I'm confused by what is going on here. If using a QVM, we remove the Quil-T, but if we are not we compile to native gates? These don't naturally feel like alternatives to me. I think it would make more sense to compile first, and if using QVM, then strip the quilT.

Reply:
Agree. Is this test code checked for correctness programmatically? Because the invocation of the compiler in the "else" case would I think fail, again because the program contains Quil-T (which is not stripped in the else case).

The normal workflow that should be represented is:
1. Start with a logical program, that excludes Quil-T.
2. If targeting a real QPU, compile via Quil-C. This produces native Quil, without Quil-T. You may also opt to start here if you are authoring a hardware-native circuit directly.
3. If you wish to control the pulse-level timing, update the Quil with Quil-T. This can include DELAY, but also DEFCAL, etc. Note: under a variant of the workflow that never requires to strip Quil-T, you also check here you are targeting a QPU, and only add the Quil-T in the first place if it is. Step (4) then becomes "run" with no special checking.
4. Run; if you are targeting a QVM, strip any Quil-T beforehand. For the QVM specifically, this does not alter the logical outcome of the simulation, so provides some defense against code which includes pulse level timing without guards but then wants to run it on a QVM. The cases where this is the right thing to do are limited, and met by the very specific feature set of the QVM, but there's no particular reason not to support stripping it as a convenience.

Reply:
Yeah, `quil_to_native_quil` was the wrong method to call there. It should be `native_quil_to_executable`. The idea is, if you have a program containing Quil-T, then remove it before running if you are targeting the QVM, otherwise you are targeting the QPU and you should translate the program with `native_quil_to_executable`. I'll fix this example.